tests/src/rules/extensions.js

```diff
@@ -359,5 +379,48 @@ ruleTester.run('extensions', rule, {
         },
       ],
     }),
+
+    test({
+      code: `
+        const foo = require('./foo.js')
+        const bar = require('./bar.json')
+        const Component = require('./Component')
+        const baz = require('foo/baz')
+        const express = require('express')
+      `,
+      options: [ 'always', {commonjs: true, ignorePackages: true} ],
+      errors: [
+        {
+          message: 'Missing file extension for "./Component"',
+          line: 4,
+          column: 27,
+        }, {
+          message: 'Missing file extension for "foo/baz"',
+          line: 5,
+          column: 21,
+        },
+      ],
+    }),
+
+    test({
+      code: `
+        const foo = require('./foo.js')
+        const bar = require('./bar.json')
+        const Component = require('./Component.jsx')
+        const express = require('express')
```

Comment:
Let’s also include a test for a package that has an extension-like name

Reply:
Do you mean like `const dot = require('./file.with.dot')` ?

Reply:
and like `require('asn1.js')`, a real package.

Reply:
Done. Also added tests for `asn1.js` for `import` statements.